### PR TITLE
Add Pug Dependencies Handling

### LINKED ISSUE #3358: Parcel 2: Pug transformer
Create the `@parcel/transformer-pug` package in `packages/transforms/pug`. This package is a Parcel 2 transform plugin for pug. It should be based on the `PugAsset` from Parcel 1.

The `@parcel/transformer-pug` package is responsible for the following things:

* Loading config from `.pugrc`, `.pugrc.js`
* Compiling pug to HTML
* Collecting dependencies as included files so Parcel watches them

### PR DESCRIPTION
**Pull request by Drapegnik**
# ↪️ Pull Request
Resolves: https://github.com/parcel-bundler/parcel/pull/3462#discussion_r318368505
Resolves #3358

## ✔️ PR Todo

- [x] Added/updated unit tests for this change
- [x] Included links to related issues/PRs

### CODE REVIEW

File: packages/core/integration-tests/test/pug.js [hunk 1]
```diff
@@ -54,14 +54,18 @@ describe('pug', function() {
       {
         type: 'html',
         name: 'index.html',
-        assets: ['index.pug']
+        assets: ['index.pug'],
+        connectedFiles: {
+          'index.pug': ['package.json', 'base.pug', 'other.pug', 'nested.pug']
```
[the remark this answers is not in the document]
Comment (Drapegnik) [reply packages/core/integration-tests/test/pug.js]: @kwelch, I think nearest `package.json` (`/parcel/packages/core/integration-tests/package.json`) is added as config file

**Comment by mkg20001** [comment]
Any updates on this?

File: packages/core/integration-tests/test/integration/pug-include-extends/index.pug [hunk 2]
```diff
@@ -2,3 +2,4 @@ extends ./base.pug
 
 block content
   include other.pug
+  include /absolute/nested.pug
```
Comment (devongovett) [inline packages/core/integration-tests/test/integration/pug-include-extends/index.pug]: interesting. how does pug resolve that? is it absolute to the filesystem or some other dir?

Comment (Drapegnik) [reply packages/core/integration-tests/test/integration/pug-include-extends/index.pug]: - `basedir: string`
> The root directory of all absolute inclusion.

**Pull request merged**